drv/lpc55-rng/src/main.rs

```diff
@@ -197,14 +225,91 @@ impl NotificationHandler for Lpc55RngServer {
     }
 }
 
+/// Load a type implementing HandoffData (and others) from a config::DataRegion.
+/// Errors will be reported in the ringbuf and will return None.
+#[cfg(feature = "dice-seed")]
+fn load_data_from_region<
+    T: for<'a> Deserialize<'a> + HandoffData + SerializedSize,
+>(
+    region: &DataRegion,
+) -> Option<T> {
+    use core::slice;
+
+    // Safety: This memory is setup by code executed before hubris and
```

Comment:
This safety comment isn't actually correct, it assumes that `region` is valid (but the caller can pass any region it likes -- the fields are public). This function should probably be unsafe in its current form, since it's basically equivalent to `ptr::read`.

You could make it safe if the caller couldn't pass any arbitrary region -- maybe hardcode which region, or identify it with an enum, or....

Reply:
Yeah that's bad. Thanks for the recommendations. I went w/ hiding the details of the memory regions behind an enum. Seems to do what you suggest: [cf2704f](https://github.com/oxidecomputer/hubris/pull/1820/commits/cf2704fae6fab72f6178fb6ca768fc171c7820d0)

Reply:
I didn't realize how badly this change affected stack usage ... working on that now.

Reply:
The majority of the stack bloat from the change in [cf2704f](https://github.com/oxidecomputer/hubris/pull/1820/commits/cf2704fae6fab72f6178fb6ca768fc171c7820d0) is mitigated in [bf092d3](https://github.com/oxidecomputer/hubris/pull/1820/commits/bf092d3ff4dbdfa4cd6a1f12b53eb74d2c9dc0b5).